webui/src/pages/repositories/repository/fileRenderers/index.tsx

```diff
@@ -1,96 +1,342 @@
+                    No rows returned.
+                </p>
+            )
+        } else {
+            const totalRows = data.length
+            let res = data;
+            if (totalRows > 100) {
+                res = data.slice(0, 100)
+            }
+            content = (
+                <>
+                    <Table striped bordered hover size={"sm"} responsive={"sm"}>
+                        <thead>
+                        <tr>
+                            {Object.getOwnPropertyNames(res[0]).map(name =>
+                                <th key={name}>{name}</th>
+                            )}
+                        </tr>
+                        </thead>
+                        <tbody>
+                        {res.map((row, i) => (
+                            <tr key={`row-${i}`}>
+                                {Object.getOwnPropertyNames(res[0]).map(name => (
+                                    <DataRow key={`row-${i}-${name}`} value={row[name]}/>
+                                ))}
+                            </tr>
+                        ))}
+                        </tbody>
+                    </Table>
+                    {(res.length < data.length) &&
+                        <Alert>{`Showing only the first ${res.length} rows (out of ${data.length})`}</Alert>
+                    }
+                </>
+            )
         }
-    }, []);
+    }
 
-    if (contentBlob) {
-        blobUrl = URL.createObjectURL(contentBlob)
-        return (
-            <img src={blobUrl} />
-        );
+
+    return (
+        <div>
+            <Form onSubmit={(e) => {
+                e.preventDefault()
+                setShouldSubmit(!shouldSubmit)
+            }}>
+                <Form.Group className="mb-2 mt-2" controlId="objectQuery">
+                    <Form.Control as="textarea" className="object-viewer-sql-input" rows={5} value={query} spellCheck={false} onChange={e => {
+                        setQuery(e.target.value)
+                    }} />
+                    <Form.Text className="text-muted align-right">
+                        Powered by <a href="https://duckdb.org/2021/10/29/duckdb-wasm.html" target="_blank" rel="noreferrer">DuckDB-WASM</a>.
+                        For a full SQL reference, see the <a href="https://duckdb.org/docs/sql/statements/select" target="_blank" rel="noreferrer">DuckDB Documentation</a>
+                    </Form.Text>
+                </Form.Group>
+                <Button type="submit" variant={"success"} >
+                    <DatabaseIcon/>{' '}
+                    Execute
+                </Button>
+            </Form>
+            <div className={"mt-3 object-viewer-sql-results"}>
+                {content}
+            </div>
+        </div>
+    )
+}
+
+interface RowProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    value: any;
+}
+
+const DataRow: FC<RowProps> = ({ value }) => {
```

Comment:
Since `RowProps` is a very minimal interface and it's probably not going to be reused, you can inline the type def.
```suggestion
const DataRow: FC<{ value: any }> = ({ value }) => {
```

Reply:
Done :)